internal/mongo/client.go

```diff
@@ -0,0 +1,174 @@
+package mongo
+
+import (
+	"context"
+	"time"
+
+	"github.com/gol4ng/logger"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type DriverCursor interface {
+	Decode(val interface{}) error
+	Next(ctx context.Context) bool
+	Close(ctx context.Context) error
+}
+
+type Option func(*client)
+
+type Client interface {
+	Replay(ctx context.Context, collection CollectionAdapter) (chan *WatchItem, error)
+	Watch(ctx context.Context, collection CollectionAdapter) (chan *WatchItem, error)
```

Comment:
Because we have 2 execution mode.

- Replay => read all the collection and push all documents in a kafka topic
- Watch => read the change stream from mongo collection and push the event in a kafka topic

Maybe we can just have one method in client interface and 2 implementations (one for replay and one for watch).
Using composition to share the common base code between the 2 implementations.

Reply:
Thanks, I will have a look at this to have 2 different implementations and only 1 method in this interface, so we could pass it as the worker and no more 2 same methods in the worker class too

Reply:
It's done: I just splitted the client in 2 parts: Replayer and Watcher